### ISSUE TITLE
Feature: Detect if plane is currently in viewport or not

### ISSUE THREAD

**Issue by ajayns** [martinlaxenaire/curtainsjs#30]
From the documentation, I assumed `plane.visible` will do the trick but it doesn't: it always shows up as `true` by default. It seems like `_shouldDraw` is what I'm looking for, but I feel like that could be exposed in the API itself.

The common use case where that will come handy is animations triggered once by visibility in the viewport: like an on appear animation. So it'll be nice to know what's already in the viewport and rest of them can be animated using `plane.onReEnterView`.

**Comment by martinlaxenaire** [comment]
Hi @ajayns,

The `visible` property should be used mainly to stop drawing/start drawing again a plane that should normally be drawn (ie is not culled). If you're used to three.js it's basically the same property as Object3D's visible property: https://threejs.org/docs/#api/en/core/Object3D.visible

I think what you are trying to achieve would look like this:

```javascript
const plane = this.curtains.addPlane(el, params);

if (plane) {
  plane.onReady(function() {
    plane.userData.isVisible = plane._shouldDraw;

    if(plane.userData.isVisible) {
      // do your appear animation
    }
  }.onReEnterView(function() {
    // plane is drawn again
    if(!plane.userData.isVisible) {
      // do your appear animation
    }

    plane.userData.isVisible = true;
  });
}
```

Judging from your issue #29 and this one I feel however that the doc and/or API are somewhat unclear about all this.
A public method to retrieve the _shouldDraw value via the API may be a good idea?

Thanks!

Cheers,

**Comment by ajayns** [comment]
Thanks for the snippet that's exactly what I was looking for. Inferring from that the `plane.userData` is where I can store all my custom data?
And, yes, it's a bit confusing I'd previously assumed parameters can be modified after initialization of plane - maybe a note could be added about that in the docs. Also, a public method/flag indicating if the plane is currently drawn could prove very useful especially in the case of animations comparable to the intersection observer ones with DOM nodes.

**Comment by martinlaxenaire** [comment]
Hey @ajayns,

`plane.userData` is indeed made to store all custom data.

I should definitely indicates which properties are read only vs the ones you can change at runtime.

Both `onLeaveView` and `onReEnterView` methods were added to handle those cases, but I will add a `isVisible` method in the next release.
For now, you can safely use this snippet, that's basically how it will be implemented in the library:

```javascript
function isPlaneVisible(plane) {
  return plane._canDraw && plane.visible && plane._shouldDraw;
}
```

Cheers,

**Comment by ajayns** [comment]
That works perfectly @martinlaxenaire 

That solves my problems. You could close this issue if you wish or wait until next release before closing so that other people with the same requirements can see this easier.

Thank you!

**Comment by martinlaxenaire** [comment]
Fixed in v5.3:
- Added a plane isDrawn() method
- Updated the docs to highlight read only properties
